### ISSUE TITLE
Ethereals can suicide and revive with the janny key to clean whatever they walk on

### ISSUE THREAD

**Issue by Pumpkinoe** [tgstation/tgstation#58171]
## Reproduction:

Suicide with janitor cart key as ethereal
Get revived by crystal 
You are now a demigod of cleaning and will clean whatever you walk over

**Comment by ATH1909** [comment]
it's... not quite that simple.

**Comment by ATH1909** [comment]
the process is as follows:
1. Grind to apprentice level in the cleaning skill.
2. Suicide with the janicart key.
3. Within the next 5 seconds, initiate a suicide with an spray bottle, then abort your spray bottle suicide attempt by interrupting its 3 second do_after(). Alternatively, perform a shame suicide.
4. Fall over dead due to the janicart key suicide that you started.
5. Be revived via a method of your choosing. I chose to use the ethereal crystal self-revive, but a defib would probably work as well (and not give you a permanent brain trauma).
6. Run around and automatically clean the tiles you run over like a janiborg.

**Comment by ATH1909** [comment]
https://youtu.be/2aBmSxvt-iU

**Comment by ATH1909** [comment]
Why this works:
The janicart key suicide returns MANUAL_SUICIDE, so it marks you as a suicider, but tells the game that it can handle actually killing you on its own. It also adds the cleaning element to you so that you can imitate being a janicart for a while (the amount of time you have before you die is dependent on your cleaning skill level).

Starting a spray bottle suicide marks you as a suicider. Aborting a spray bottle suicide returns MANUAL_SUICIDE_NONLETHAL (it's one of the few suicides left in the game that still do so), which clears your suicider status (it's technically a variable, but hush). Shame suicides do a similar thing, but also give you a strong negative moodlet.

When your allotted time to have fun with the janicart key suicide is up, you die, but the cleaning element is not removed from you. Your suicider status is NOT set to true again here, and since the spray bottle suicide set your suicider status to false earlier, you can be revived.

Once you've been revived (and probably even while you're still dead, if someone drags you around or something), you can continue to use your newfound janicart powers that janiborgs can get 10 seconds into the round.

**Comment by ATH1909** [comment]
Unfortunately, this is the most practical application of this technique that I've found. The soda can suicide's sleep()s can allow you to delay being unmarked as a suicider by a few seconds, allowing you to use this technique even with instant suicides. However, the "payoff" suicide still needs to be a manual suicide* that doesn't ghost or gib you, and the only suicides in the game that meet that criteria (that I could find) are the janicart suicide, the picture of the virgin mary suicide, and the toy singularity suicide under certain conditions. The janicart suicide can actually be useful, but the portrait of the virgin mary just makes you say a few lines before killing you and the toy singularity suicide's only useful feature (I mean, moving the toy singularity into you could technically be useful, but the signaler suicide does a similar thing in a much simpler/less risky way) already unmarks you as a suicider when it's triggered.

*This is because the code for suicides will automatically kill and ghost you on its own if MANUAL_SUICIDE, MANUAL_SUICIDE_NONLETHAL, or SHAME isn't returned, and none of the entries in the latter two categories gain any new uses from this trick, as far as I can tell.

**Comment by ATH1909** [comment]
tldr;
this trick is only useful for pretending to be a janicart for longer than you're intended to, as far as my extensive research can tell.

thank you for coming to my ted talk.

**Comment by ATH1909** [comment]
Note to self: Test to see if replica pods could be a way around the ghosting problem.

**Comment by ArcaneDefence** [comment]
All this convinces me is that I shouldn't have left the element as is when they die because I thought the idea of someone dragging a corpse around to clean things up was funny.
